conans/test/functional/build_requires/profile_requires_test.py

```diff
@@ -236,5 +263,9 @@ def test_profile_requires(self):
 cmake
 gcc
 """
-        self.client.save({"profile.txt": profile, "conanfile.txt": reuse}, clean_first=True)
-        self.client.run("install . --profile ./profile.txt --build missing")
+        self.client.save({"profile_host.txt": profile_host,
```

Comment:
My concern when I see a modified test is: Or we are breaking the behavior or we are not testing something anymore. I think this is case 2). Shouldn't we provide new tests but keeping the classic `--profile=whatever?`. Am I missing something?

Reply:
Some tests are changed because of the breaking change (1) explained in the PR notes, before the final release that change has to be reverted and many tests will have no changes: for the release, `build_requires` should use the `host` profile too unless the user explicitly says the opposite.

But right now we are far from thinking about the release and these tests show how the `host` and `build` profiles behave for different graph scenarios, it is important to check that the context switch is done and every package is getting the expected settings, options and environment variables.